Mathlib/Data/List/BigOperators/Basic.lean

```diff
@@ -79,105 +89,103 @@ theorem prod_repeat (a : M) (n : ℕ) : (repeat a n).Prod = a ^ n :=
 #align list.prod_repeat List.prod_repeat
 
 @[to_additive sum_eq_card_nsmul]
-theorem prod_eq_pow_card (l : List M) (m : M) (h : ∀ x ∈ l, x = m) : l.Prod = m ^ l.length := by
-  rw [← prod_repeat, ← list.eq_repeat.mpr ⟨rfl, h⟩]
+theorem prod_eq_pow_card (l : List M) (m : M) (h : ∀ x ∈ l, x = m) : l.prod = m ^ l.length := by
+  rw [← prod_replicate, ← List.eq_replicate.mpr ⟨rfl, h⟩]
 #align list.prod_eq_pow_card List.prod_eq_pow_card
 
 @[to_additive]
 theorem prod_hom_rel (l : List ι) {r : M → N → Prop} {f : ι → M} {g : ι → N} (h₁ : r 1 1)
-    (h₂ : ∀ ⦃i a b⦄, r a b → r (f i * a) (g i * b)) : r (l.map f).Prod (l.map g).Prod :=
+    (h₂ : ∀ ⦃i a b⦄, r a b → r (f i * a) (g i * b)) : r (l.map f).prod (l.map g).prod :=
   List.recOn l h₁ fun a l hl => by simp only [map_cons, prod_cons, h₂ hl]
 #align list.prod_hom_rel List.prod_hom_rel
 
 @[to_additive]
 theorem prod_hom (l : List M) {F : Type _} [MonoidHomClass F M N] (f : F) :
-    (l.map f).Prod = f l.Prod :=
+    (l.map f).prod = f l.prod :=
   by
-  simp only [Prod, foldl_map, ← map_one f]
-  exact l.foldl_hom _ _ _ 1 (map_mul f)
+  simp only [prod, foldl_map, ← map_one f]
+  exact l.foldl_hom f (. * .) (. * f .) 1 (fun x y => (map_mul f x y).symm)
 #align list.prod_hom List.prod_hom
 
 @[to_additive]
 theorem prod_hom₂ (l : List ι) (f : M → N → P) (hf : ∀ a b c d, f (a * b) (c * d) = f a c * f b d)
     (hf' : f 1 1 = 1) (f₁ : ι → M) (f₂ : ι → N) :
-    (l.map fun i => f (f₁ i) (f₂ i)).Prod = f (l.map f₁).Prod (l.map f₂).Prod :=
+    (l.map fun i => f (f₁ i) (f₂ i)).prod = f (l.map f₁).prod (l.map f₂).prod :=
   by
-  simp only [Prod, foldl_map]
-  convert l.foldl_hom₂ (fun a b => f a b) _ _ _ _ _ fun a b i => _
-  · exact hf'.symm
-  · exact hf _ _ _ _
+  simp only [prod, foldl_map]
+  rw [← l.foldl_hom₂ (fun a b => f a b), hf']
```

Comment:
Maybe add a porting note about the broken `convert` since this comes up a lot and we'll want to fix it eventually.